jsettlers.logic/src/jsettlers/logic/movable/MovableStrategy.java

```diff
@@ -45,10 +46,12 @@
 public abstract class MovableStrategy implements Serializable {
 	private static final long serialVersionUID = 3135655342562634378L;
 
-	private final Movable movable;
+	protected final Movable movable;
+	protected final AbstractMovableGrid grid;
```

Comment:
I don't think that is cleaner to have redundant data.

Don't worry about that method call. This is Java, not C++. We have a compiler to worry about that ;-).